### ISSUE TITLE
Create veg_soil_bgc_interface src subdirectory and have Veg-CN and soilbiogeochemistry interface modules in it

### ISSUE THREAD

**Issue by ekluzek** [ESCOMP/CTSM#1992]
Currently biogeochem and soilbiogeochem are separated, but have a set of modules that have both and provide an interface between the two. There are also some lower level modules that are deeper down and are exclusive to either biogeochem or soilbiogeochem. It would be helpful to make this design more obvious by making a subdirectory that contained the interfaces between the two, and being more explicit about which modules act as interfaces between the two and which really are only part of one or the other.

The other reason for doing this is so that FATES can share the soil-biogeochemistry calling sequence without having to have it's own copy of it. The soil-BGC calling sequence is complex enough that sharing this code makes sense. Right now if soil-BGC is changed, changes have to go into both FATES and the biogeochem calling sequences. 

Doing this also makes the current design more obvious and gives an easy rule to know where to add modules (if they contain both biogeochem and soilbiogeochem it goes in veg_soil_vgc_interface and if it's confined to only one or the other it goes in the relevent one).

**Comment by ekluzek** [comment]
@billsacks @adrifoster and others any thoughts on the proposed name here for the subdirectory?

I don't think this means we should have everything in this subdirectory start with veg_soil_bgc_interface, as that sounds awkward. But, they probably should move away from starting with either CN or SoilBiogeochem at least for the files in this subdirectory.

**Comment by billsacks** [comment]
I think this is a great idea. No initial thoughts on the name.

**Comment by ekluzek** [comment]
Some utils to move here:

CNSharedParamsMod rename to VegSoilBGCParams for

CNParamsShareInst, nlev_soildecomp_standard, use_fun

SoilBGC:
SoilBiogeochemCompetitionMod.F90
SoilBiogeochemDecompCascadeMIMICSMod.F90

VegBGC_and_soilBGC_Summary:CNVegNitrogenStateType.F90
   currently: CNVegCarbonStateType.F90 and 
  could be in SoilBiogeochemCarbonStateType.F90 and SoilBiogeochemNitrogenStateType.F90



BGC:
ch4Mod.F90
CNBalanceCheckMod.F90
CNC14DecayMod.F90
CNCIsoFluxMod.F90
CNCStateUpdate1Mod.F90
CNCStateUpdate2Mod.F90
CNCStateUpdate3Mod.F90
CNDriverMod.F90
CNFireBaseMod.F90
CNFireLi2014Mod.F90
CNFireLi2016Mod.F90
CNFireLi2021Mod.F90
CNFUNMod.F90
CNGapMortalityMod.F90
CNNDynamicsMod.F90
CNNStateUpdate1Mod.F90
CNNStateUpdate2Mod.F90
CNNStateUpdate3Mod.F90
CNRootDynMod.F90
CNVegCarbonFluxType.F90
CNVegCarbonStateType.F90
CNVegetationFacade.F90
CNVegNitrogenFluxType.F90
CNVegNitrogenStateType.F90
dynConsBiogeochemMod.F90
dynGrossUnrepMod.F90
FATESFireBase.F90
NutrientCompetitionCLM45defaultMod.F90
NutrientCompetitionFlexibleCNMod.F90
NutrientCompetitionMethodMod.F90

**Comment by rgknox** [comment]
@ekluzek I think you capture the 4 key tags in your proposed folder name "BGC" "veg" "soil" "interface". So some permutation of those, truncated or not makes sense to me.  

Am I correct in understanding that  CNVegetationFacade and CN_driver are the two existing modules that do most of the heavy  lifting on interfacy stuff, and that they would live in the new folder? And that perhaps those modules would be renamed and massaged as well?

**Comment by ekluzek** [comment]
@rgknox yes, I list the modules I think should live there above, and they include those two. Yeah, and I think they should have name changes. They are both complicated and it would be nice if the path that FATES took in them was reasonably obvious, so I think doing some changes to make them clearer would be good.

**Comment by wwieder** [comment]
At the SE meeting we were generally supportive of this idea, but don't want to hold up #1959 in making this happen.  As such, we can work on this design somewhat more incrementally.

**Comment by ekluzek** [comment]
I think this discussion applies here:

https://github.com/ESCOMP/CTSM/pull/1959/files#r1168090937

**Comment by ekluzek** [comment]
Another discussion that applies:

https://github.com/ESCOMP/CTSM/pull/1959/files/79a77d6eb3e3ca1b16e8abb29a68b844221bc4ee#r1167680622
